### DISCUSSION TITLE
Unable to send a POST login request using Cypress

### DISCUSSION THREAD

**Discussion by adriantan09** [cypress-io/cypress#17933] [Questions and Help]
I am writing an end-to-end test for a website using Cypress and would like to make an API POST request to authenticate a user.

I have been able to make a request that sends back a 200 response code with an appropriate jwt.

Here is a code snippet of what the request looks like:

`Cypress.Commands.add('login', () => {

cy.request({
    method: 'POST',
    url: 'https://api.website.com/api/login',
    body: {
        email: "xxx",
        password: "xxx"
    }
}).then((resp) => {
    window.localStorage.setItem('jwt', resp.body.token)
    // console.log(resp.body.token)
    cy.log(resp)
})
cy.visit('/')`

Here is an image of the Cypress test output:
https://i.stack.imgur.com/LUufs.png 

Here is an image of the Chrome console, showing a successful response code:
https://i.stack.imgur.com/zhBPi.png

After reading the [Cypress command log](https://docs.cypress.io/guides/guides/network-requests#Command-Log), I noticed that the white circle indicates the request was unable to reach the server. What are the possible reasons for this happening? I did some further debugging and realised that regardless of the request type I make (GET, POST, etc) the request is not being fulfilled.

Thanks in advance!

**Comment by daniel-simoes-koerber** [comment]
Multiple things can be wrong here: 
1. Is your API up and running? 
2. Are you able to perform the same request on another tool, such as postman? Do you obtain the exact same response, as in cypress? (also using cy.log may not be the best idea - use instead `console.log` to show the full response) 
3. From the attachments, we can see that you got a 200 OK. This usually means that everything is working as it's supposed :) So, were you expecting a different result? 
4. (My own opinion) do not focus too much on the UI filled or not filled circle to say that your request was not fullfilled. It can simply be a UI bug

In summary from all that you provided seems to me that you were able to perform the request with success and it's simply a bug on the UI that is telling you that you were not able to do it (even if your logs say the opposite). And if this is true, then an issue should be open 

Hope that I could help :)

**Reply by adriantan09** [reply]
Thank you for your reply. I have managed to solve my issue. The request was working however, I was not setting the right data in local storage in order to authenticate. After using the Google Chrome Developer tools, I found that both the auth._token.local and auth.refresh_token.local needed to be set after the request. I have attached the code snippet below to help anyone who comes across a similar issue. Thanks.

cy.request({
        method: 'POST',
        url: 'http://127.0.0.1:8000/api/login',
        body: {
            email: "email",
            password: "password",
        }
    }).then((resp) => {
        localStorage.setItem('auth._token.local', 'Bearer ' + resp.body.token)
        localStorage.setItem('auth._refresh_token.local', false)
    });
    cy.visit('/');
